eth/state_accessor.go

```diff
@@ -212,6 +212,7 @@ func (eth *Ethereum) stateAtTransaction(block *types.Block, txIndex int, reexec
 		msg, _ := tx.AsMessage(signer)
 		txContext := core.NewEVMTxContext(msg)
 		context := core.NewEVMBlockContext(block.Header(), eth.blockchain, nil)
+		statedb.Prepare(tx.Hash(), block.Hash(), idx)
```

Comment:
I think we should move this down (so that it does *not* prepare the statedb for the tx to be traced), and rather do it on call-site. That way all 4 invocations of `traceTx` will be the same. In the current code 3 manually does a prep and one inherits it from here, which seems brittle.